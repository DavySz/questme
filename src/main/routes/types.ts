export interface IRoute {
  name: string;
  component: () => React.ReactNode;
}
