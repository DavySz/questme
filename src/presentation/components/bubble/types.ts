import { ReactNode } from "react";

export interface IPositionWrapperStyles {
  top: number;
  left: number;
  right: number;
  bottom: number;
}

export interface IBubbleRoot extends Partial<IPositionWrapperStyles> {
  children: ReactNode;
}

export interface IBubbleCircle {
  size: number;
}

export interface ICircleStyles extends IBubbleCircle {}
