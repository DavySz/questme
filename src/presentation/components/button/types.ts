import { TThemeColors } from "@presentation/styles";
import { ElementType } from "react";
import { TouchableOpacityProps } from "react-native";

export type TButtonVariant =
  | "primary"
  | "secondary"
  | "tertiary"
  | "google"
  | "facebook";

export type TIconSide = "left" | "right" | "none";

export interface IButtonRoot extends TouchableOpacityProps {
  variant?: TButtonVariant;
  children: React.ReactNode;
  full?: boolean;
}

export interface IButtonText {
  variant?: TButtonVariant;
  children: React.ReactNode;
}

export interface IRootStyle {
  variant: TButtonVariant;
  full: boolean;
}

export interface IChildProps {
  variant: TButtonVariant;
}

export interface IButtonIcon {
  icon: ElementType;
  side: TIconSide;
  color?: TThemeColors;
}

export interface IButtonIconStyles {
  side: TIconSide;
}
