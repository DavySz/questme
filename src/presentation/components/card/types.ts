import { TThemeColors } from "@presentation/styles";
import { ElementType, ReactNode } from "react";

export interface ICardRoot {
  children: ReactNode;
}

export interface ICardGroup {
  children: ReactNode;
}

export interface ICardIcon {
  icon: ElementType;
  color?: TThemeColors;
}
