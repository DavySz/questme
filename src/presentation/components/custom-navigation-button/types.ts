import { BottomTabBarButtonProps } from "@react-navigation/bottom-tabs";

import { TThemeColors } from "@presentation/styles";
import { ElementType, ReactNode } from "react";

export interface ICustomNavigationIcon {
  icon: ElementType;
  color?: TThemeColors;
}

export interface ICustomNavigationButtonRoot extends BottomTabBarButtonProps {
  children: ReactNode;
}
