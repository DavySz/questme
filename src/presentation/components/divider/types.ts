export interface IDividerText {
  children: React.ReactNode;
}
