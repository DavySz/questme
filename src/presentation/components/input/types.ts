import { TThemeColors } from "@presentation/styles";
import { ElementType, ReactNode } from "react";
import { TextInputProps } from "react-native";

export type TIconSide = "left" | "right";

export interface IInputIcon {
  icon: ElementType;
  color?: TThemeColors;
}

export interface IInputRoot {
  children: ReactNode;
}

export interface IChildProps {
  handleBlur: () => void;
  handleFocus: () => void;
}

export interface IInputWrapperStyles {
  isFocused: boolean;
}

export interface IInputLabel {
  children: ReactNode;
}

export interface IInputText extends TextInputProps, Partial<IChildProps> {}
