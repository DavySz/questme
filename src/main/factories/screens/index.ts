export * from "./onboarding-factory";
export * from "./splash-factory";
export * from "./initial-factory";
export * from "./login-factory";
export * from "./sign-up-factory";
