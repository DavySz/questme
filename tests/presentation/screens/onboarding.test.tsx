import { Onboarding } from "@presentation/screens/onboarding/onboarding";
import { OnboardingDot } from "@presentation/screens/onboarding/onboarding.dot";
import { slides } from "@presentation/screens/onboarding/slides";
import { ISlide } from "@presentation/screens/onboarding/types";
import Illustration from "@presentation/assets/illustrations/onboarding-first-illustration.svg";
import { OnboardingContainer } from "@presentation/screens";
import { customRender, TestingPublicNavigator } from "../../utils";
import { fireEvent, render } from "@testing-library/react-native";

const mockSlides: ISlide[] = [
  {
    id: 1,
    description: "slide1",
    image: <Illustration />,
  },
  {
    id: 2,
    description: "slide2",
    image: <Illustration />,
  },
  {
    id: 3,
    description: "slide3",
    image: <Illustration />,
  },
];

describe("Onboarding", () => {
  describe("Onboarding Container", () => {
    test("Should render onboarding container without errors", () => {
      const tree = customRender(<OnboardingContainer />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe("OnboardingDot", () => {
    test("Should render correct dot component when isSelected", () => {
      const { getByTestId } = customRender(<OnboardingDot isSelected />);

      const dotSelected = getByTestId("is-selected-dot");

      expect(dotSelected).toBeTruthy();
    });

    test("Should render correct dot component when isSelected is false", () => {
      const { getByTestId } = customRender(
        <OnboardingDot isSelected={false} />
      );

      const dotSelected = getByTestId("is-not-selected-dot");

      expect(dotSelected).toBeTruthy();
    });
  });

  describe("slides", () => {
    test("Should slides have 3 items", () => {
      expect(slides.length).toBe(3);
    });
  });

  describe("Onboarding", () => {
    test("Should slides have 3 items in FlatList component", () => {
      const updateCurrentSlideIndexJest = jest.fn();
      const handleGoToSignUpScreenJest = jest.fn();
      const handleGoToLoginScreenJest = jest.fn();

      const { getByTestId } = customRender(
        <Onboarding
          currentIndex={0}
          slides={mockSlides}
          handleGoToLoginScreen={handleGoToLoginScreenJest}
          handleGoToSignUpScreen={handleGoToSignUpScreenJest}
          updateCurrentSlideIndex={updateCurrentSlideIndexJest}
        />
      );

      const flatList = getByTestId("slides-list");
      expect(flatList.props.data.length).toBe(mockSlides.length);
    });

    test("Should navigate to sign up screen when button is clicked", async () => {
      const { getByTestId, findByTestId } = render(
        <TestingPublicNavigator currentRoute="onboarding" />
      );

      const buttonResetPassword = getByTestId("button-sign-up");
      fireEvent.press(buttonResetPassword);

      const resetPasswordScreen = await findByTestId("sign-up-screen-content");

      expect(resetPasswordScreen).toBeTruthy();
    });

    test("Should navigate to login screen when button is clicked", async () => {
      const { getByTestId, findByTestId } = render(
        <TestingPublicNavigator currentRoute="onboarding" />
      );

      const buttonResetPassword = getByTestId("button-login");
      fireEvent.press(buttonResetPassword);

      const resetPasswordScreen = await findByTestId("login-screen-content");

      expect(resetPasswordScreen).toBeTruthy();
    });
  });
});
