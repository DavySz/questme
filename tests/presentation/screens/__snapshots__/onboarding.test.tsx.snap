// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding Onboarding Container Should render onboarding container without errors 1`] = `
<View
  style={
    {
      "backgroundColor": "#6A5AE0",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
  testID="bubbles-container"
>
  <View
    left={56}
    style={
      {
        "left": 56,
        "position": "absolute",
        "top": 174,
      }
    }
    top={174}
  >
    <View
      size={48}
      style={
        {
          "backgroundColor": "#C9F2E9",
          "borderBottomLeftRadius": 24,
          "borderBottomRightRadius": 24,
          "borderTopLeftRadius": 24,
          "borderTopRightRadius": 24,
          "height": 48,
          "opacity": 0.1,
          "width": 48,
        }
      }
      type="fill"
    />
  </View>
  <View
    bottom={200}
    right={78}
    style={
      {
        "bottom": 200,
        "position": "absolute",
        "right": 78,
      }
    }
  >
    <View
      size={20}
      style={
        {
          "backgroundColor": "#C9F2E9",
          "borderBottomLeftRadius": 10,
          "borderBottomRightRadius": 10,
          "borderTopLeftRadius": 10,
          "borderTopRightRadius": 10,
          "height": 20,
          "opacity": 0.1,
          "width": 20,
        }
      }
      type="fill"
    />
  </View>
  <View
    bottom={0}
    left={-121}
    style={
      {
        "left": -121,
        "position": "absolute",
      }
    }
  >
    <View
      size={242}
      style={
        {
          "backgroundColor": "#C9F2E9",
          "borderBottomLeftRadius": 121,
          "borderBottomRightRadius": 121,
          "borderTopLeftRadius": 121,
          "borderTopRightRadius": 121,
          "height": 242,
          "opacity": 0.1,
          "width": 242,
        }
      }
      type="fill"
    />
  </View>
  <View
    bottom={-83.6}
    left={-209}
    style={
      {
        "bottom": -83.6,
        "left": -209,
        "position": "absolute",
      }
    }
  >
    <View
      size={418}
      style={
        {
          "backgroundColor": "transparent",
          "borderBottomLeftRadius": 209,
          "borderBottomRightRadius": 209,
          "borderColor": "#C9F2E9",
          "borderStyle": "solid",
          "borderTopLeftRadius": 209,
          "borderTopRightRadius": 209,
          "borderWidth": 1,
          "height": 418,
          "opacity": 0.1,
          "width": 418,
        }
      }
      type="outline"
    />
  </View>
  <View
    right={-121}
    style={
      {
        "position": "absolute",
        "right": -121,
      }
    }
    top={0}
  >
    <View
      size={242}
      style={
        {
          "backgroundColor": "#C9F2E9",
          "borderBottomLeftRadius": 121,
          "borderBottomRightRadius": 121,
          "borderTopLeftRadius": 121,
          "borderTopRightRadius": 121,
          "height": 242,
          "opacity": 0.1,
          "width": 242,
        }
      }
      type="fill"
    />
  </View>
  <View
    right={-209}
    style={
      {
        "position": "absolute",
        "right": -209,
        "top": -83.6,
      }
    }
    top={-83.6}
  >
    <View
      size={418}
      style={
        {
          "backgroundColor": "transparent",
          "borderBottomLeftRadius": 209,
          "borderBottomRightRadius": 209,
          "borderColor": "#C9F2E9",
          "borderStyle": "solid",
          "borderTopLeftRadius": 209,
          "borderTopRightRadius": 209,
          "borderWidth": 1,
          "height": 418,
          "opacity": 0.1,
          "width": 418,
        }
      }
      type="outline"
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-end",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      }
    }
    variant="tertiary"
  >
    <RCTScrollView
      bounces={false}
      data={
        [
          {
            "description": "Create gamified quizzes becomes simple",
            "id": 0,
            "image": <SvgMock />,
          },
          {
            "description": "Find quizzes to test out your knowledge",
            "id": 1,
            "image": <SvgMock />,
          },
          {
            "description": "Take part in challenges with friends",
            "id": 2,
            "image": <SvgMock />,
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsHorizontalScrollIndicator={false}
      stickyHeaderIndices={[]}
      testID="slides-list"
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <View
              style={
                {
                  "alignSelf": "center",
                }
              }
            >
              <SvgMock />
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <View
              style={
                {
                  "alignSelf": "center",
                }
              }
            >
              <SvgMock />
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <View
              style={
                {
                  "alignSelf": "center",
                }
              }
            >
              <SvgMock />
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 12,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#6A5AE0",
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderColor": "#FFF",
            "borderStyle": "solid",
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
            "borderWidth": 2,
            "height": 16,
            "justifyContent": "center",
            "width": 16,
          }
        }
        testID="is-selected-dot"
      >
        <View
          style={
            {
              "backgroundColor": "#FFF",
              "borderBottomLeftRadius": 4,
              "borderBottomRightRadius": 4,
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
              "height": 8,
              "width": 8,
            }
          }
        />
      </View>
      <View
        style={
          {
            "backgroundColor": "#FFF",
            "borderBottomLeftRadius": 4,
            "borderBottomRightRadius": 4,
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
            "height": 8,
            "width": 8,
          }
        }
        testID="is-not-selected-dot"
      />
      <View
        style={
          {
            "backgroundColor": "#FFF",
            "borderBottomLeftRadius": 4,
            "borderBottomRightRadius": 4,
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
            "height": 8,
            "width": 8,
          }
        }
        testID="is-not-selected-dot"
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderBottomLeftRadius": 20,
          "borderBottomRightRadius": 20,
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "gap": 24,
          "justifyContent": "center",
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": "100%",
        }
      }
    >
      <Text
        align="center"
        color="neutral-black"
        style={
          {
            "color": "#0C092A",
            "fontFamily": "Rubik_500Medium",
            "fontSize": 24,
            "lineHeight": 36,
            "textAlign": "center",
          }
        }
        variant="heading-3"
      >
        Create gamified quizzes becomes simple
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "gap": 16,
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#6A5AE0",
              "borderBottomLeftRadius": 20,
              "borderBottomRightRadius": 20,
              "borderTopLeftRadius": 20,
              "borderTopRightRadius": 20,
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
              "width": "100%",
            }
          }
          testID="button-root-wrapper"
        >
          <Text
            color="neutral-white"
            style={
              {
                "color": "#FFF",
                "fontFamily": "Rubik_500Medium",
                "fontSize": 16,
                "lineHeight": 24,
              }
            }
            variant="body-normal-medium"
          >
            Sign Up
          </Text>
        </View>
        <Text
          color="neutral-gey-2"
          style={
            {
              "color": "#858494",
              "fontFamily": "Rubik_400Regular",
              "fontSize": 16,
              "lineHeight": 24,
            }
          }
          variant="body-normal-regular"
        >
          Already have an account?
           
          <Text
            color="primary"
            style={
              {
                "color": "#6A5AE0",
                "fontFamily": "Rubik_500Medium",
                "fontSize": 16,
                "lineHeight": 24,
              }
            }
            variant="body-normal-medium"
          >
            Login
          </Text>
        </Text>
      </View>
    </View>
  </View>
</View>
`;
